app/invocation/invocation_suggestion_card.tsx

```diff
@@ -327,23 +327,56 @@ ${yamlSuggestions.map((s) => `      ${s}`).join("\n")}`}
     if (model.optionsMap.get("remote_cache_compression")) return null;
     if (model.optionsMap.get("experimental_remote_cache_compression")) return null;
     if (!model.optionsMap.get("remote_cache") && !model.optionsMap.get("remote_executor")) return null;
-    const version = getBazelMajorVersion(model);
+
+    const versions = getBazelVersion(model);
     // Bazel pre-v5 doesn't support compression.
-    if (version === null || version < 5) return null;
+    if (versions === null || versions.major < 5) return null;
+
+    const flag = versions.major >= 7 ? "--experimental_remote_cache_compression" : "--remote_cache_compression";
 
     return {
       level: SuggestionLevel.INFO,
       message: (
         <>
-          Consider adding the Bazel flag <BazelFlag>--experimental_remote_cache_compression</BazelFlag> to improve
-          remote cache throughput.
+          Consider adding the Bazel flag <BazelFlag>{flag}</BazelFlag> to improve remote cache throughput.
         </>
       ),
       reason: (
         <>
-          Shown because this build is cache-enabled but{" "}
-          <span className="inline-code">--experimental_remote_cache_compression</span> is neither enabled nor explicitly
-          disabled.
+          Shown because this build is cache-enabled but <span className="inline-code">{flag}</span> is neither enabled
+          nor explicitly disabled.
+        </>
+      ),
+    };
+  },
+  ({ model }) => {
+    if (!capabilities.config.expandedSuggestionsEnabled) return null;
+    if (!model.isBazelInvocation()) return null;
+
+    if (model.optionsMap.get("experimental_remote_cache_compression_threshold")) return null;
+    if (
+      !model.optionsMap.get("remote_cache_compression") &&
+      !model.optionsMap.get("experimental_remote_cache_compression")
+    )
+      return null;
+    if (!model.optionsMap.get("remote_cache") && !model.optionsMap.get("remote_executor")) return null;
+
+    const versions = getBazelVersion(model);
+    // threshold flag is available from Bazel 7.1 forward
+    if (versions === null || versions.major < 7 || versions.minor < 1) return null;
+
+    return {
+      level: SuggestionLevel.INFO,
+      message: (
+        <>
+          Consider adding the Bazel flag <BazelFlag>--experimental_remote_cache_compression_threshold=100</BazelFlag> to
```

Comment:
wdyt about doing some sampling of real blobs before we recommend an exact value here - I know 100 is the value we've chosen on the server but the roundness of the number makes me think it was chosen somewhat arbitrarily?

Reply:
I built our repo (buildbuddy.git) locally with `bazel build //... --disk_cache=~/.bazel/disk_cache`.

Then ran `fd -S '-200B' -x zstd {}` to find and compress all files under 200 bytes.
The output of `zstd` is then trimmed and piped through `sort -n | uniq -c`, which gives this output.

```
~/.bazel/disk_cache> cat temp.txt| sort -n | uniq -c
   2  55.10%   (   196 B =>    108 B
   1  56.32%   (   190 B =>    107 B
   1  59.30%   (   199 B =>    118 B
   1  59.59%   (   193 B =>    115 B
   1  59.68%   (   186 B =>    111 B
   1  59.89%   (   187 B =>    112 B
   1  59.90%   (   197 B =>    118 B
   1  60.10%   (   193 B =>    116 B
   1  60.10%   (   198 B =>    119 B
   1  60.11%   (   188 B =>    113 B
   1  60.21%   (   191 B =>    115 B
   1  60.32%   (   189 B =>    114 B
   2  60.41%   (   197 B =>    119 B
   5  60.51%   (   195 B =>    118 B
   1  60.53%   (   190 B =>    115 B
   1  60.62%   (   193 B =>    117 B
   1  60.71%   (   196 B =>    119 B
   1  60.73%   (   191 B =>    116 B
   1  60.91%   (   197 B =>    120 B
   1  60.94%   (   192 B =>    117 B
   1  60.96%   (   187 B =>    114 B
   1  61.03%   (   195 B =>    119 B
   2  61.08%   (   185 B =>    113 B
   1  61.14%   (   175 B =>    107 B
   1  61.14%   (   193 B =>    118 B
   1  61.17%   (   188 B =>    115 B
   1  61.20%   (   183 B =>    112 B
   1  61.26%   (   191 B =>    117 B
   1  61.33%   (   181 B =>    111 B
   2  61.34%   (   194 B =>    119 B
   1  61.36%   (   176 B =>    108 B
   1  61.38%   (   189 B =>    116 B
   1  61.41%   (   184 B =>    113 B
   2  61.50%   (   187 B =>    115 B
   1  61.54%   (   182 B =>    112 B
   1  61.54%   (   195 B =>    120 B
   1  61.58%   (   190 B =>    117 B
   1  61.59%   (   138 B =>     85 B
   1  61.62%   (   185 B =>    114 B
   3  61.66%   (   193 B =>    119 B
   1  61.71%   (   175 B =>    108 B
   1  61.73%   (   196 B =>    121 B
   1  61.75%   (   183 B =>    113 B
   6  61.78%   (   191 B =>    118 B
   2  61.81%   (   199 B =>    123 B
   1  61.83%   (   186 B =>    115 B
   1  61.85%   (   173 B =>    107 B
   1  61.86%   (   194 B =>    120 B
   2  61.88%   (   181 B =>    112 B
   4  61.90%   (   189 B =>    117 B
   1  61.93%   (   197 B =>    122 B
   1  61.98%   (   192 B =>    119 B
   1  62.00%   (   200 B =>    124 B
   2  62.03%   (   187 B =>    116 B
   4  62.05%   (   195 B =>    121 B
   1  62.11%   (   190 B =>    118 B
   5  62.12%   (   198 B =>    123 B
   1  62.15%   (   177 B =>    110 B
   2  62.16%   (   185 B =>    115 B
   2  62.18%   (   193 B =>    120 B
   1  62.22%   (   180 B =>    112 B
   4  62.24%   (   196 B =>    122 B
   3  62.29%   (   175 B =>    109 B
   1  62.30%   (   183 B =>    114 B
   2  62.30%   (   191 B =>    119 B
   3  62.31%   (   199 B =>    124 B
   4  62.36%   (   178 B =>    111 B
   3  62.37%   (   194 B =>    121 B
   1  62.43%   (   181 B =>    113 B
   3  62.43%   (   189 B =>    118 B
   5  62.44%   (   197 B =>    123 B
   1  62.50%   (   176 B =>    110 B
   4  62.50%   (   184 B =>    115 B
   2  62.50%   (   192 B =>    120 B
   2  62.50%   (   200 B =>    125 B
   3  62.56%   (   195 B =>    122 B
   2  62.57%   (   179 B =>    112 B
   2  62.57%   (   187 B =>    117 B
   1  62.63%   (   198 B =>    124 B
   1  62.64%   (   182 B =>    114 B
   2  62.71%   (   177 B =>    111 B
   1  62.78%   (   180 B =>    113 B
   2  62.83%   (   191 B =>    120 B
   2  62.84%   (   183 B =>    115 B
   1  62.87%   (   167 B =>    105 B
   2  62.89%   (   194 B =>    122 B
   1  62.90%   (   186 B =>    117 B
   2  62.92%   (   178 B =>    112 B
   3  62.94%   (   197 B =>    124 B
   2  62.96%   (   189 B =>    119 B
   1  62.98%   (   181 B =>    114 B
   1  63.00%   (   200 B =>    126 B
   1  63.01%   (   173 B =>    109 B
   4  63.02%   (   192 B =>    121 B
   1  63.07%   (   176 B =>    111 B
   1  63.08%   (   195 B =>    123 B
   4  63.10%   (   187 B =>    118 B
   1  63.13%   (   179 B =>    113 B
   3  63.13%   (   198 B =>    125 B
   3  63.16%   (   190 B =>    120 B
   2  63.21%   (   193 B =>    122 B
   4  63.22%   (   174 B =>    110 B
   5  63.24%   (   185 B =>    117 B
   1  63.27%   (   196 B =>    124 B
   2  63.28%   (   177 B =>    112 B
   4  63.30%   (   188 B =>    119 B
   2  63.32%   (   199 B =>    126 B
   1  63.35%   (   191 B =>    121 B
   6  63.43%   (   175 B =>    111 B
   2  63.45%   (   197 B =>    125 B
   3  63.48%   (   178 B =>    113 B
   2  63.49%   (   189 B =>    120 B
   2  63.58%   (   173 B =>    110 B
   2  63.59%   (   184 B =>    117 B
   3  63.64%   (   187 B =>    119 B
   1  63.69%   (   168 B =>    107 B
   1  63.73%   (   193 B =>    123 B
   1  63.79%   (   174 B =>    111 B
   1  63.87%   (   191 B =>    122 B
   2  63.89%   (   180 B =>    115 B
   3  63.93%   (   183 B =>    117 B
   2  63.96%   (   197 B =>    126 B
   1  63.98%   (   186 B =>    119 B
   1  64.00%   (   175 B =>    112 B
   1  64.13%   (   184 B =>    118 B
   1  64.20%   (   176 B =>    113 B
   1  64.25%   (   179 B =>    115 B
   1  64.25%   (   193 B =>    124 B
   1  64.29%   (   182 B =>    117 B
   1  64.37%   (   174 B =>    112 B
   1  64.41%   (   177 B =>    114 B
   2  64.44%   (   180 B =>    116 B
   1  64.48%   (   183 B =>    118 B
   1  64.55%   (   189 B =>    122 B
   1  64.62%   (   195 B =>    126 B
   2  65.62%   (   192 B =>    126 B
   1  67.24%   (   116 B =>     78 B
   1  70.62%   (   177 B =>    125 B
   1  70.92%   (   196 B =>    139 B
   1  72.16%   (   194 B =>    140 B
   3  72.25%   (   191 B =>    138 B
   1  72.96%   (   159 B =>    116 B
   4  73.74%   (   179 B =>    132 B
   1  73.82%   (   191 B =>    141 B
   1  74.19%   (   186 B =>    138 B
   1  75.27%   (   182 B =>    137 B
   4  75.74%   (   169 B =>    128 B
   1  75.89%   (   141 B =>    107 B
   1  76.38%   (   199 B =>    152 B
   1  77.06%   (   170 B =>    131 B
   1  77.71%   (   157 B =>    122 B
   3  78.76%   (   193 B =>    152 B
   1  79.08%   (   153 B =>    121 B
   2  79.41%   (   136 B =>    108 B
   1  80.25%   (   162 B =>    130 B
   1  80.27%   (   147 B =>    118 B
   1  80.49%   (   164 B =>    132 B
   1  80.81%   (   172 B =>    139 B
   1  81.72%   (   186 B =>    152 B
   1  81.73%   (   197 B =>    161 B
   1  81.77%   (   181 B =>    148 B
   2  81.82%   (   165 B =>    135 B
   1  82.00%   (   150 B =>    123 B
   1  82.52%   (   143 B =>    118 B
   9  82.64%   (   144 B =>    119 B
   1  82.84%   (   169 B =>    140 B
   1  82.95%   (   176 B =>    146 B
  12  83.22%   (   143 B =>    119 B
   1  83.24%   (   179 B =>    149 B
   1  83.25%   (   197 B =>    164 B
 217  83.33%   (   144 B =>    120 B
   1  83.61%   (   183 B =>    153 B
   1  83.78%   (   148 B =>    124 B
  47  83.92%   (   143 B =>    120 B
 636  84.03%   (   144 B =>    121 B
   1  84.25%   (   146 B =>    123 B
   7  84.35%   (   147 B =>    124 B
  19  84.46%   (   148 B =>    125 B
   2  84.52%   (   168 B =>    142 B
  88  84.62%   (   143 B =>    121 B
1898  84.72%   (   144 B =>    122 B
   1  84.97%   (   153 B =>    130 B
  86  85.03%   (   147 B =>    125 B
   1  85.13%   (   195 B =>    166 B
 303  85.14%   (   148 B =>    126 B
  83  85.31%   (   143 B =>    122 B
3225  85.42%   (   144 B =>    123 B
   2  85.45%   (   165 B =>    141 B
   4  85.52%   (   145 B =>    124 B
   2  85.53%   (   152 B =>    130 B
   1  85.62%   (   153 B =>    131 B
 248  85.71%   (   147 B =>    126 B
 838  85.81%   (   148 B =>    127 B
   2  85.91%   (   149 B =>    128 B
   1  85.96%   (   114 B =>     98 B
   2  85.98%   (   107 B =>     92 B
 126  86.01%   (   143 B =>    123 B
4214  86.11%   (   144 B =>    124 B
   1  86.14%   (   166 B =>    143 B
  18  86.18%   (   152 B =>    131 B
   1  86.21%   (   145 B =>    125 B
   2  86.27%   (   153 B =>    132 B
 413  86.39%   (   147 B =>    127 B
2655  86.49%   (   148 B =>    128 B
   6  86.58%   (   149 B =>    129 B
  92  86.71%   (   143 B =>    124 B
4838  86.81%   (   144 B =>    125 B
  54  86.84%   (   152 B =>    132 B
   2  86.90%   (   145 B =>    126 B
   8  86.93%   (   153 B =>    133 B
 502  87.07%   (   147 B =>    128 B
3920  87.16%   (   148 B =>    129 B
  16  87.25%   (   149 B =>    130 B
  57  87.41%   (   143 B =>    125 B
   1  87.50%   (   144 B =>
5266  87.50%   (   144 B =>    126 B
  62  87.50%   (   152 B =>    133 B
  18  87.58%   (   153 B =>    134 B
   4  87.59%   (   145 B =>    127 B
   1  87.67%   (   146 B =>    128 B
 587  87.76%   (   147 B =>    129 B
5919  87.84%   (   148 B =>    130 B
  25  87.92%   (   149 B =>    131 B
  25  88.11%   (   143 B =>    126 B
  62  88.16%   (   152 B =>    134 B
2464  88.19%   (   144 B =>    127 B
  12  88.24%   (   153 B =>    135 B
   2  88.28%   (   145 B =>    128 B
 627  88.44%   (   147 B =>    130 B
7393  88.51%   (   148 B =>    131 B
  29  88.59%   (   149 B =>    132 B
   1  88.70%   (   177 B =>    157 B
  10  88.81%   (   143 B =>    127 B
  31  88.82%   (   152 B =>    135 B
 694  88.89%   (   144 B =>    128 B
  22  88.89%   (   153 B =>    136 B
   3  88.89%   (   198 B =>    176 B
   2  88.94%   (   199 B =>    177 B
 510  89.12%   (   147 B =>    131 B
4804  89.19%   (   148 B =>    132 B
  38  89.26%   (   149 B =>    133 B
   5  89.27%   (   177 B =>    158 B
   3  89.39%   (   198 B =>    177 B
   3  89.45%   (   199 B =>    178 B
  19  89.47%   (   152 B =>    136 B
   3  89.51%   (   143 B =>    128 B
   7  89.54%   (   153 B =>    137 B
 567  89.58%   (   144 B =>    129 B
   1  89.58%   (   192 B =>    172 B
   2  89.60%   (   173 B =>    155 B
   1  89.74%   (   195 B =>    175 B
 291  89.80%   (   147 B =>    132 B
   3  89.83%   (   177 B =>    159 B
2299  89.86%   (   148 B =>    133 B
   2  89.90%   (   198 B =>    178 B
  25  89.93%   (   149 B =>    134 B
   3  89.95%   (   199 B =>    179 B
  10  90.13%   (   152 B =>    137 B
   5  90.17%   (   173 B =>    156 B
   4  90.20%   (   153 B =>    138 B
   1  90.21%   (   143 B =>    129 B
 148  90.28%   (   144 B =>    130 B
   1  90.29%   (   175 B =>    158 B
   1  90.36%   (   197 B =>    178 B
   5  90.40%   (   177 B =>    160 B
   1  90.43%   (   188 B =>    170 B
 112  90.48%   (   147 B =>    133 B
 852  90.54%   (   148 B =>    134 B
   7  90.60%   (   149 B =>    135 B
   1  90.62%   (   192 B =>    174 B
   9  90.75%   (   173 B =>    157 B
   2  90.77%   (   195 B =>    177 B
   3  90.79%   (   152 B =>    138 B
   1  90.86%   (   175 B =>    159 B
   1  90.91%   (   198 B =>    180 B
   1  90.95%   (   199 B =>    181 B
   3  90.96%   (   177 B =>    161 B
   2  90.97%   (   144 B =>    131 B
   1  91.01%   (   189 B =>    172 B
   1  91.03%   (   156 B =>    142 B
  39  91.16%   (   147 B =>    134 B
 473  91.22%   (   148 B =>    135 B
   2  91.28%   (   149 B =>    136 B
   4  91.33%   (   173 B =>    158 B
   1  91.41%   (   198 B =>    181 B
   2  91.43%   (   175 B =>    160 B
   1  91.44%   (   187 B =>    171 B
   1  91.45%   (   152 B =>    139 B
   2  91.50%   (   200 B =>    183 B
   6  91.53%   (   177 B =>    162 B
   1  91.74%   (   109 B =>    100 B
  16  91.84%   (   147 B =>    135 B
   1  91.84%   (   196 B =>    180 B
   2  91.88%   (   197 B =>    181 B
 103  91.89%   (   148 B =>    136 B
   1  91.95%   (   149 B =>    137 B
   1  91.98%   (   162 B =>    149 B
   1  92.00%   (   200 B =>    184 B
   2  92.09%   (   177 B =>    163 B
   2  92.17%   (   166 B =>    153 B
   1  92.26%   (   168 B =>    155 B
   1  92.31%   (   195 B =>    180 B
   1  92.35%   (   196 B =>    181 B
   1  92.47%   (   186 B =>    172 B
   2  92.52%   (   107 B =>     99 B
   6  92.52%   (   147 B =>    136 B
   1  92.57%   (   148 B =>    137 B
   1  92.63%   (   190 B =>    176 B
   1  92.67%   (   191 B =>    177 B
   1  92.71%   (   192 B =>    178 B
   1  92.89%   (   197 B =>    183 B
   1  92.96%   (   199 B =>    185 B
   1  93.02%   (   129 B =>    120 B
   1  93.28%   (   119 B =>    111 B
   1  93.55%   (   124 B =>    116 B
   1  93.59%   (   156 B =>    146 B
  12  93.70%   (   127 B =>    119 B
  31  93.80%   (   129 B =>    121 B
   1  94.02%   (   184 B =>    173 B
  21  94.49%   (   127 B =>    120 B
  32  94.57%   (   129 B =>    122 B
  13  95.28%   (   127 B =>    121 B
   2  95.35%   (   129 B =>    123 B
   1  95.36%   (   194 B =>    185 B
   1  95.97%   (   149 B =>    143 B
   4  96.00%   (   125 B =>    120 B
  20  96.06%   (   127 B =>    122 B
   1  96.13%   (   181 B =>    174 B
   1  96.85%   (   127 B =>    123 B
   1  96.88%   (    64 B =>     62 B
   1  96.91%   (    97 B =>     94 B
   1  97.22%   (   108 B =>    105 B
   5  97.60%   (   125 B =>    122 B
   5  97.67%   (   129 B =>    126 B
   1  98.19%   (   166 B =>    163 B
   1  99.41%   (   169 B =>    168 B
   1 100.00%   (    64 B =>     64 B
   1 100.00%   (    76 B =>     76 B
   1 100.00%   (   125 B =>    125 B
   2 100.00%   (   129 B =>    129 B
   1 100.00%   (   163 B =>    163 B
   1 100.00%   (   165 B =>    165 B
   1 100.00%   (   168 B =>    168 B
   1 100.66%   (   151 B =>    152 B
   1 100.66%   (   152 B =>    153 B
   1 101.83%   (   109 B =>    111 B
   1 101.96%   (   153 B =>    156 B
   1 102.20%   (    91 B =>     93 B
   1 103.92%   (   102 B =>    106 B
   1 104.46%   (   157 B =>    164 B
   1 104.52%   (   155 B =>    162 B
   1 104.61%   (   152 B =>    159 B
   1 105.61%   (   107 B =>    113 B
   1 106.54%   (   107 B =>    114 B
   1 107.07%   (    99 B =>    106 B
   2 107.27%   (   110 B =>    118 B
   3 107.34%   (   109 B =>    117 B
   2 107.41%   (   108 B =>    116 B
   1 107.48%   (   107 B =>    115 B
   1 108.00%   (   100 B =>    108 B
   2 108.18%   (   110 B =>    119 B
   1 108.23%   (   158 B =>    171 B
   3 108.26%   (   109 B =>    118 B
   1 108.86%   (    79 B =>     86 B
   1 109.41%   (    85 B =>     93 B
   1 109.57%   (    94 B =>    103 B
   1 109.70%   (   134 B =>    147 B
   1 111.43%   (    35 B =>     39 B
   1 111.82%   (   110 B =>    123 B
   5 111.93%   (   109 B =>    122 B
   3 112.62%   (   103 B =>    116 B
   1 112.75%   (   102 B =>    115 B
   1 113.04%   (    69 B =>     78 B
   3 113.27%   (    98 B =>    111 B
   2 113.54%   (    96 B =>    109 B
   4 113.68%   (    95 B =>    108 B
   7 113.83%   (    94 B =>    107 B
   3 113.98%   (    93 B =>    106 B
   4 114.29%   (    91 B =>    104 B
   3 114.61%   (    89 B =>    102 B
   1 114.77%   (    88 B =>    101 B
   7 114.94%   (    87 B =>    100 B
   4 115.12%   (    86 B =>     99 B
   2 115.66%   (    83 B =>     96 B
   1 115.69%   (    51 B =>     59 B
   1 116.46%   (    79 B =>     92 B
   1 120.97%   (    62 B =>     75 B
   1 121.31%   (    61 B =>     74 B
   2 122.03%   (    59 B =>     72 B
   1  126 B
   2 126.53%   (    49 B =>     62 B
   1 129.41%   (    34 B =>     44 B
   1 130.95%   (    42 B =>     55 B
   3 132.50%   (    40 B =>     53 B
   1 134.21%   (    38 B =>     51 B
   1 139.39%   (    33 B =>     46 B
   9 140.62%   (    32 B =>     45 B
   1 143.33%   (    30 B =>     43 B
   1 181.25%   (    16 B =>     29 B
   1 192.86%   (    14 B =>     27 B
   1 218.18%   (    11 B =>     24 B
   1 244.44%   (     9 B =>     22 B
   2 285.71%   (     7 B =>     20 B
   5 360.00%   (     5 B =>     18 B
   1 425.00%   (     4 B =>     17 B
   1 533.33%   (     3 B =>     16 B
   1 1400.00%   (     1 B =>     14 B
```

So I think it's a bit hard to come up with an exact number.
Depending on the content of the original file, it could be compressed better or worse.
But generally, it's pretty clear that starting from 150B, the smaller the file, the easier it is to inflate after compression. I think 100B is kinda a sweet spot.

Wdyt? 🤔

Reply:
The 100B number we chose on the server side was due to these comments:

![Screenshot 2024-06-28 at 2 23 01 PM](https://github.com/user-attachments/assets/7c6fa946-8140-44ac-9a12-bfbef2dac5d3)

Reply:
yeah, based on that data, and after looking at a few invocations, 100 LGTM